[GTK] Remove WebKitWebContext:process-swap-on-cross-site-navigation-enabled and ensure PSON is always enabled with GTK 4

mcatanzaro opened the pull request:
#### 94be12a42fa1ad0b89aba0e4bb55943c182f86bf
<pre>
[GTK] Remove WebKitWebContext:process-swap-on-cross-site-navigation-enabled and ensure PSON is always enabled with GTK 4
<a href="https://bugs.webkit.org/show_bug.cgi?id=220116">https://bugs.webkit.org/show_bug.cgi?id=220116</a>

Reviewed by Carlos Garcia Campos.

In WPE, PSON is mandatory and always enabled. We cannot do this for GTK
3 without breaking existing apps, but we can do it for GTK 4.

* Source/WebKit/UIProcess/API/glib/WebKitWebContext.cpp:
(webkitWebContextConstructed):
(webkit_web_context_class_init):
* Tools/MiniBrowser/gtk/main.c:
(activate):
* Tools/TestWebKitAPI/glib/WebKitGLib/TestMain.h:
(Test::Test):

Canonical link: <a href="https://commits.webkit.org/255825@main">https://commits.webkit.org/255825@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a648cefe22cf646035287b0a80aff350ba870f93

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/93725 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/2920 "Built successfully") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/43/builds/24335 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/103369 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/163690 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/97718 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/2935 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/31176 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/86064 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/99409 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/99385 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/2088 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/80165 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/29115 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/84013 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/43/builds/24335 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/72068 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/37569 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/43/builds/24335 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/35418 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/43/builds/24335 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4025 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/39296 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/41387 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/41230 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/43/builds/24335 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
<!--EWS-Status-Bubble-End-->

carlosgcampos commented on Source/WebKit/UIProcess/API/glib/WebKitWebContext.cpp:
```diff
@@ -417,11 +417,9 @@ static void webkitWebContextConstructed(GObject* object)
     configuration.setInjectedBundlePath(FileSystem::stringFromFileSystemRepresentation(bundleFilename.get()));
     configuration.setUsesWebProcessCache(true);
-#if PLATFORM(GTK)
+#if PLATFORM(GTK) && !USE(GTK4)
     configuration.setProcessSwapsOnNavigation(priv->psonEnabled);
-#if !USE(GTK4)
     configuration.setUseSystemAppearanceForScrollbars(priv->useSystemAppearanceForScrollbars);
-#endif
```
Don't you need to set it to true for GTK4?

mcatanzaro replied:
Nope, that's the default set in WebPreferencesExperimental.yaml. I got confused by this too yesterday and had to check to be sure PSON was really enabled for WPE.

carlosgcampos replied:
Even for non developer builds? I still think it's safer to explicitly set it to true here.

mcatanzaro replied:
> Even for non developer builds?

I think so? I don't think the runtime features are hooked up to `ENABLE(DEVELOPER_MODE)`.

But we can explicitly set it, sure. It should hopefully not change anything. If it does, then PSON will be enabled for WPE for the first time.

mcatanzaro replied:
OK, done. Better?

carlosgcampos replied:
> I think so? I don't think the runtime features are hooked up to ENABLE(DEVELOPER_MODE).

I thought preferences in WebPreferencesExperimental were only enabled when `ENABLE(EXPERIMENTAL_FEATURES)`, but they aren't. In any case, it doesn't hurt to make it explicit, it's clearer and won't break if the preferences change in the future.

carlosgcampos commented on Source/WebKit/UIProcess/API/glib/WebKitWebContext.cpp:
```diff
@@ -417,11 +409,12 @@ static void webkitWebContextConstructed(GObject* object)
-#endif
+#else
+    configuration.setProcessSwapsOnNavigation(true);
+    configuration.setUseSystemAppearanceForScrollbars(false);
```
This broke the wpe build, setUseSystemAppearanceForScrollbars is GTK specific

mcatanzaro replied:
Sigh, fixed....

mcatanzaro committed:
[GTK] Remove WebKitWebContext:process-swap-on-cross-site-navigation-enabled and ensure PSON is always enabled with GTK 4
https://bugs.webkit.org/show_bug.cgi?id=220116

Reviewed by Carlos Garcia Campos.

In WPE, PSON is mandatory and always enabled. We cannot do this for GTK
3 without breaking existing apps, but we can do it for GTK 4.

* Source/WebKit/UIProcess/API/glib/WebKitWebContext.cpp:
(webkitWebContextConstructed):
(webkit_web_context_class_init):
* Tools/MiniBrowser/gtk/main.c:
(activate):
* Tools/TestWebKitAPI/glib/WebKitGLib/TestMain.h:
(Test::Test):

Canonical link: https://commits.webkit.org/255825@main
